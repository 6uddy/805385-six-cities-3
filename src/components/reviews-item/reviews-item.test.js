import React from 'react';
import renderer from 'react-test-renderer';
import ReviewsItem from "./reviews-item.jsx";

const review = {
  id: 0,
  comment: `The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.`,
  date: new Date(2020, 1, 1, 0, 0, 0, 0).toISOString(),
  rating: 3.5,
  user: {
    id: 0,
    name: `Max`,
    avatarUrl: `/img/avatar-max.jpg`,
    isPro: true
  },
};

it(`Test ReviewsItem component`, () => {

  const tree = renderer
    .create(
        <ReviewsItem
          review={review}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
