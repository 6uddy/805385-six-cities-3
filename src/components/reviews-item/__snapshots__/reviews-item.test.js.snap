// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ReviewsItem component 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="/img/avatar-max.jpg"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Max
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "70%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.
    </p>
    <time
      className="reviews__time"
      dateTime="2020-02-01"
    >
      February 2020
    </time>
  </div>
</li>
`;
